### ISSUE TITLE
LiquidHaskell is causing HLS to fails 

### ISSUE THREAD

**Issue by ashkann** [haskell/haskell-language-server#3134]
vscode 1.71.0 with Haskell plugin configured to use ghcup +  almost zero code stack project
MacOS
ghc 8.10.7
stack 2.7.5 
HLS 1.7.0.0

Trying to make LiquidHaskell work with the HLS. The `stack build` works so LH setup is well but now LHS outputs an error

```
No 'hie.yaml' found. Try to discover the project type!
Run entered for haskell-language-server-wrapper(haskell-language-server-wrapper) Version 1.7.0.0 x86_64 ghc-9.2.2
Current directory: /Users/ashkan/Hobby/Haskell/learn-lh
Operating system: darwin
Arguments: ["--lsp","-d"]
Cradle directory: /Users/ashkan/Hobby/Haskell/learn-lh
Cradle type: Stack

Tool versions found on the $PATH
cabal:		3.6.2.0
stack:		2.7.5
ghc:		8.10.7


Consulting the cradle to get project GHC version...
Project GHC version: 8.10.7
haskell-language-server exe candidates: ["haskell-language-server-8.10.7","haskell-language-server"]
Launching haskell-language-server exe at:/Users/ashkan/.ghcup/tmp/ghcup-ghc-8.10.7_cabal-3.6.2.0_hls-1.7.0.0_stack-2.7.5/haskell-language-server-8.10.7
Failed to get project GHC executable path: CradleError {cradleErrorDependencies = [], cradleErrorExitCode = ExitFailure 1, cradleErrorStderr = ["Error when calling stack exec ghc -- -v0 -package-env=- -ignore-dot-ghci -e Control.Monad.join (Control.Monad.fmap System.IO.putStr System.Environment.getExecutablePath)","","\n<interactive>:0:1: error:\n    Not in scope: \8216Control.Monad.join\8217\n    No module named \8216Control.Monad\8217 is imported.\n\n<interactive>:0:21: error:\n    Not in scope: \8216Control.Monad.fmap\8217\n    No module named \8216Control.Monad\8217 is imported.\n\n<interactive>:0:40: error:\n    Not in scope: \8216System.IO.putStr\8217\n    No module named \8216System.IO\8217 is imported.\n\n<interactive>:0:57: error:\n    Not in scope: \8216System.Environment.getExecutablePath\8217\n    No module named \8216System.Environment\8217 is imported.\n"]}
[Error - 11:13:30 PM] Connection to server got closed. Server will not be restarted.
[Error - 11:13:30 PM] Notify file events failed.
```

I tried the command on terminal, same error :
```
learn-lh  $> stack exec ghc -- -v0 -package-env=- -ignore-dot-ghci -e "Control.Monad.join (Control.Monad.fmap System.IO.putStr System.Environment.getExecutablePath)"

<interactive>:0:1: error:
    Not in scope: ‘Control.Monad.join’
    No module named ‘Control.Monad’ is imported.

<interactive>:0:21: error:
    Not in scope: ‘Control.Monad.fmap’
    No module named ‘Control.Monad’ is imported.

<interactive>:0:40: error:
    Not in scope: ‘System.IO.putStr’
    No module named ‘System.IO’ is imported.

<interactive>:0:57: error:
    Not in scope: ‘System.Environment.getExecutablePath’
    No module named ‘System.Environment’ is imported.
```

The minimum to reproduce is to simply add the `liquid-base` (should replace the `base`) :
```
dependencies:
# - base
- liquid-base
# - liquid-vector
# - liquid-containers
# - liquid-bytestring
- liquidhaskell  
```
if I comment out the `liquid-base` and put back the `base` then HLS works but LH integration is crippled (it needs the `liquid-base` to work properly).

the`stack.yaml` file, taken verbatim from `lh-plugin-demo` https://github.com/ucsd-progsys/lh-plugin-demo :
```
allow-newer: true
resolver: lts-18.27
compiler: ghc-8.10.7
packages:
  - .
extra-deps:
  - git: https://github.com/ucsd-progsys/liquidhaskell
    commit: a4f2198f894810a729feb0f8d06301790ee8d1fb
    subdirs:
      - .
      - liquid-base
      - liquid-prelude
      - liquid-ghc-prim
      # - liquid-containers
      # - liquid-bytestring
      # - liquid-vector
  - git: https://github.com/ucsd-progsys/liquid-fixpoint
    commit: 08fa6b6e0001c2da6b96a135523ca56107a3c0d6
  - git: https://github.com/facundominguez/rest
    commit: 31e974979c90e910efe5199ee0d3721b791667f6
```

`package.yaml` file :
```
name:                learn-lh
version:             0.1.0.0
github:              "githubuser/learn-lh"
license:             BSD3
author:              "Author name here"
maintainer:          "example@example.com"
copyright:           "2022 Author name here"

extra-source-files:
- README.md
- CHANGELOG.md
description:         Please see the README on GitHub at <https://github.com/githubuser/learn-lh#readme>

dependencies:
# - base
- liquid-base
# - liquid-vector
# - liquid-containers
# - liquid-bytestring
- liquidhaskell

ghc-options:
- -Wall
- -Wcompat
- -Widentities
- -Wincomplete-record-updates
- -Wincomplete-uni-patterns
- -Wmissing-export-lists
- -Wmissing-home-modules
- -Wpartial-fields
- -Wredundant-constraints
- -fdefer-type-errors -fplugin=LiquidHaskell

library:
  source-dirs:
    - src

executables:
  learn-lh-exe:
    main:                Main.hs
    source-dirs:         app
    ghc-options:
    - -threaded
    - -rtsopts
    - -with-rtsopts=-N
    dependencies:
    - learn-lh

tests:
  learn-lh-test:
    main:                Spec.hs
    source-dirs:         test
    ghc-options:
    - -threaded
    - -rtsopts
    - -with-rtsopts=-N
    dependencies:
    - learn-lh
 ```
 
 Thanks.

**Comment by ashkann** [comment]
I spent some more time on it. Running the main executable works fine but the wrapper results in the exact same error :
```
Users/ashkan/.ghcup/tmp/ghcup-ghc-8.10.7_cabal-3.6.2.0_hls-1.7.0.0_stack-2.7.5/haskell-language-server-wrapper-1.7.0.0 
Found "/Users/ashkan/Hobby/Haskell/learn-lh/hie.yaml" for "/Users/ashkan/Hobby/Haskell/learn-lh/a"
Run entered for haskell-language-server-wrapper(haskell-language-server-wrapper-1.7.0.0) Version 1.7.0.0 x86_64 ghc-9.2.2
Current directory: /Users/ashkan/Hobby/Haskell/learn-lh
Operating system: darwin
Arguments: []
Cradle directory: /Users/ashkan/Hobby/Haskell/learn-lh
Cradle type: Stack

Tool versions found on the $PATH
cabal:          3.6.2.0
stack:          2.7.5
ghc:            8.10.7


Consulting the cradle to get project GHC version...
Project GHC version: 8.10.7
haskell-language-server exe candidates: ["haskell-language-server-8.10.7","haskell-language-server"]
Launching haskell-language-server exe at:/Users/ashkan/.ghcup/bin/haskell-language-server-8.10.7
Failed to get project GHC executable path: CradleError {cradleErrorDependencies = [], cradleErrorExitCode = ExitFailure 1, cradleErrorStderr = ["Error when calling stack exec ghc -- -v0 -package-env=- -ignore-dot-ghci -e Control.Monad.join (Control.Monad.fmap System.IO.putStr System.Environment.getExecutablePath)","","\n<interactive>:0:1: error:\n    Not in scope: \8216Control.Monad.join\8217\n    No module named \8216Control.Monad\8217 is imported.\n\n<interactive>:0:21: error:\n    Not in scope: \8216Control.Monad.fmap\8217\n    No module named \8216Control.Monad\8217 is imported.\n\n<interactive>:0:40: error:\n    Not in scope: \8216System.IO.putStr\8217\n    No module named \8216System.IO\8217 is imported.\n\n<interactive>:0:57: error:\n    Not in scope: \8216System.Environment.getExecutablePath\8217\n    No module named \8216System.Environment\8217 is imported.\n"]}
```

Notice that I run the wrapper `haskell-language-server-wrapper-1.7.0.0` as opposed to `haskell-language-server-8.10.7` (which runs with no errors, shows the expected output and terminates).

**Comment by fendor** [comment]
I have trouble getting a liquid Haskell project up and running, your example doesn't compile for me with stack: 

```
> stack build
Building all executables for `liquid-haskell-test' once. After a successful build of all of them, only specified executables will be rebuilt.
liquid-haskell-test> configure (exe)
Configuring liquid-haskell-test-0.1.0.0...
liquid-haskell-test> build (exe)
Preprocessing executable 'liquid-haskell-test' for liquid-haskell-test-0.1.0.0..
Building executable 'liquid-haskell-test' for liquid-haskell-test-0.1.0.0..
[1 of 1] Compiling Main

/home/hugin/Documents/haskell/liquid-haskell-test/src/Main.hs:1:8: error:
    Ambiguous module name ‘Prelude’:
      it was found in multiple packages:
      base-4.14.3.0 liquid-base-4.14.3.0
  |
1 | module Main (main) where
  |        ^^^^


--  While building package liquid-haskell-test-0.1.0.0 (scroll up to its section to see the error) using:
      /home/hugin/.stack/setup-exe-cache/x86_64-linux-tinfo6/Cabal-simple_mPHDZzAJ_3.2.1.0_ghc-8.10.7 --builddir=.stack-work/dist/x86_64-linux-tinfo6/Cabal-3.2.1.0 build exe:liquid-haskell-test --ghc-options " -fdiagnostics-color=always"
    Process exited with code: ExitFailure 1
```

Can you maybe share an example I just have to check out?

**Comment by fendor** [comment]
Maybe stack instructs GHC too eagerly to not use implicit prelude, or something? Or the package-env is cleared too vigorously?

**Comment by ashkann** [comment]
Narrowed it down to this :
```
stack exec ghc -- -e "System.Environment.getExecutablePath"

<interactive>:0:1: error:
    Not in scope: ‘System.Environment.getExecutablePath’
    No module named ‘System.Environment’ is imported.
```
and maybe even narrower :
```
stack exec ghc -- -e "import Prelude"

<no location info>: error:
    Ambiguous module name ‘Prelude’:
      it was found in multiple packages:
      base-4.14.3.0 liquid-base-4.14.3.0       
``` 

I'm basically a n00b on both Haskell and LH but the second command is rather confusing because didn't we just opted out of `base` and opted in for `liquid-base` in `stack.yaml` ?

**Comment by ashkann** [comment]
> Maybe stack instructs GHC too eagerly to not use implicit prelude, or something? Or the package-env is cleared too vigorously?

I wish I could understand what you are saying :D I'll try ... so why would stack tell GHC to not use implicit prelude ? besides in haskell-language-server-wrapper code it is explicitly mentioned the lines (that causes the error) is there to support NoImplicitPrelude 
https://github.com/haskell/haskell-language-server/blob/b5a37f7fc360596899cb2945f363030f44156415/exe/Wrapper.hs#L117

**Comment by ashkann** [comment]
Published the min reproduce on https://github.com/ashkann/stack-lh-min

- Doesn't list *any* dependencies in `package.yaml`, only in `stack.yaml`
- Doesn't actually include LH GHC plugin (no `ghc-options` in `package.yaml`) 
- `stack build` succeeds but HLS fails
```shell
/Users/ashkan/.ghcup/tmp/ghcup-ghc-8.10.7_cabal-3.6.2.0_hls-1.7.0.0_stack-2.7.5/haskell-language-server-wrapper-1.7.0.0
No 'hie.yaml' found. Try to discover the project type!
Run entered for haskell-language-server-wrapper(haskell-language-server-wrapper-1.7.0.0) Version 1.7.0.0 x86_64 ghc-9.2.2
Current directory: /Users/ashkan/Hobby/Haskell/stack-lh-min
Operating system: darwin
Arguments: []
Cradle directory: /Users/ashkan/Hobby/Haskell/stack-lh-min
Cradle type: Stack

Tool versions found on the $PATH
cabal:          3.6.2.0
stack:          2.7.5
ghc:            8.10.7


Consulting the cradle to get project GHC version...
Project GHC version: 8.10.7
haskell-language-server exe candidates: ["haskell-language-server-8.10.7","haskell-language-server"]
Launching haskell-language-server exe at:/Users/ashkan/.ghcup/bin/haskell-language-server-8.10.7
Failed to get project GHC executable path: CradleError {cradleErrorDependencies = [], cradleErrorExitCode = ExitFailure 1, cradleErrorStderr = ["Error when calling stack exec ghc -- -v0 -package-env=- -ignore-dot-ghci -e Control.Monad.join (Control.Monad.fmap System.IO.putStr System.Environment.getExecutablePath)","","\n<interactive>:0:1: error:\n    Not in scope: \8216Control.Monad.join\8217\n    No module named \8216Control.Monad\8217 is imported.\n\n<interactive>:0:21: error:\n    Not in scope: \8216Control.Monad.fmap\8217\n    No module named \8216Control.Monad\8217 is imported.\n\n<interactive>:0:40: error:\n    Not in scope: \8216System.IO.putStr\8217\n    No module named \8216System.IO\8217 is imported.\n\n<interactive>:0:57: error:\n    Not in scope: \8216System.Environment.getExecutablePath\8217\n    No module named \8216System.Environment\8217 is imported.\n"]}
```
- Also running directly on shell :
```shell
stack exec ghc -- -v0 -package-env=- -ignore-dot-ghci -e "Control.Monad.join (Control.Monad fmap System.IO.putStr System.Environment.getExecutablePath)"

<interactive>:0:1: error:
    Not in scope: ‘Control.Monad.join’
    No module named ‘Control.Monad’ is imported.

<interactive>:0:21: error:
    Not in scope: data constructor ‘Control.Monad’
    No module named ‘Control’ is imported.

<interactive>:0:40: error:
    Not in scope: ‘System.IO.putStr’
    No module named ‘System.IO’ is imported.

<interactive>:0:57: error:
    Not in scope: ‘System.Environment.getExecutablePath’
    No module named ‘System.Environment’ is imported.
```
- Making it simpler :
```shell
stack exec ghc -- -v0 -package-env=- -ignore-dot-ghci -e "System.Environment.getExecutablePath"

<interactive>:0:1: error:
    Not in scope: ‘System.Environment.getExecutablePath’
    No module named ‘System.Environment’ is imported.
```
- Removing all the flags etc and trying to import Prelude:
```shell 
stack exec ghc -- -e "import Prelude"

<no location info>: error:
    Ambiguous module name ‘Prelude’:
      it was found in multiple packages:
      base-4.14.3.0 liquid-base-4.14.3.0
```

**Comment by fendor** [comment]
Your example project seems to work for me :see_no_evil: 

![image](https://user-images.githubusercontent.com/8463814/188472268-2e87ea9f-cadc-483a-b699-994ec2e44554.png)


My logs:
```
> haskell-language-server-wrapper --debug app/
No 'hie.yaml' found. Try to discover the project type!
Run entered for haskell-language-server-wrapper(haskell-language-server-wrapper) Version 1.7.0.0 x86_64 ghc-9.2.2
Current directory: /home/hugin/Documents/haskell/stack-lh-min
Operating system: linux
Arguments: ["--debug","app/"]
Cradle directory: /home/hugin/Documents/haskell/stack-lh-min
Cradle type: Stack

Tool versions found on the $PATH
cabal:		3.6.2.0
stack:		2.7.5
ghc:		8.10.7


Consulting the cradle to get project GHC version...
Project GHC version: 8.10.7
haskell-language-server exe candidates: ["haskell-language-server-8.10.7","haskell-language-server"]
Launching haskell-language-server exe at:/home/hugin/.ghcup/bin/haskell-language-server-8.10.7
2022-09-05T14:32:46.375716Z | Info | No log file specified; using stderr.
2022-09-05T14:32:46.376381Z | Info | haskell-language-server version: 1.7.0.0 (GHC: 8.10.7) (PATH: /home/hugin/.ghcup/hls/1.7.0.0/lib/haskell-language-server-1.7.0.0/bin/haskell-language-server-8.10.7)
2022-09-05T14:32:46.376820Z | Info | Directory: /home/hugin/Documents/haskell/stack-lh-min
2022-09-05T14:32:46.377010Z | Info | Logging heap statistics every 60.00s
 ghcide setup tester in /home/hugin/Documents/haskell/stack-lh-min.
Report bugs at https://github.com/haskell/haskell-language-server/issues

Step 1/4: Finding files to test in /home/hugin/Documents/haskell/stack-lh-min
Found 1 files

Step 2/4: Looking for hie.yaml files that control setup
Found 1 cradle
  ()

Step 3/4: Initializing the IDE
2022-09-05T14:32:46.379319Z | Debug | Initializing exports map from hiedb
2022-09-05T14:32:46.379722Z | Debug | Shake session initialized

Step 4/4: Type checking the files2022-09-05T14:32:46.380271Z | Debug | Done initializing exports map from hiedb. Size: 9

2022-09-05T14:32:46.380907Z | Info | Cradle path: app/Main.hs
2022-09-05T14:32:46.380978Z | Warning | No [cradle](https://github.com/mpickering/hie-bios#hie-bios) found for app/Main.hs.
Proceeding with [implicit cradle](https://hackage.haskell.org/package/implicit-hie).
You should ignore this message, unless you see a 'Multi Cradle: No prefixes matched' error.
2022-09-05T14:32:46.383361Z | Debug | Cradle: Cradle {cradleRootDir = "/home/hugin/Documents/haskell/stack-lh-min", cradleOptsProg = CradleAction: Stack}
2022-09-05 16:32:46.697356972 [ThreadId 30] INFO hie-bios:	Using main module: 1. Package `stack-lh-min' component stack-lh-min:exe:stack-lh-min-exe with main-is file: /home/hugin/Documents/haskell/stack-lh-min/app/Main.hs
2022-09-05 16:32:46.850556841 [ThreadId 30] INFO hie-bios:	The following GHC options are incompatible with GHCi and have not been passed to it: -threaded
2022-09-05 16:32:46.850642001 [ThreadId 30] INFO hie-bios:	Configuring GHCi with the following packages: stack-lh-min
2022-09-05 16:32:47.347897295 [ThreadId 37] INFO hie-bios:	/home/hugin/Documents/haskell/stack-lh-min/.stack-work/install/x86_64-linux-tinfo6/7c444beeea79842f1b013590dc23a79fb9a40e04eb4aa3bf7bdcf007de05a149/8.10.7/pkgdb:/home/hugin/.stack/snapshots/x86_64-linux-tinfo6/7c444beeea79842f1b013590dc23a79fb9a40e04eb4aa3bf7bdcf007de05a149/8.10.7/pkgdb:/home/hugin/.stack/programs/x86_64-linux/ghc-tinfo6-8.10.7/lib/ghc-8.10.7/package.conf.d
2022-09-05T14:32:48.253228Z | Debug | Session loading result: Right (ComponentOptions {componentOptions = ["-i","-odir=/home/hugin/Documents/haskell/stack-lh-min/.stack-work/odir","-hidir=/home/hugin/Documents/haskell/stack-lh-min/.stack-work/odir","-hide-all-packages","-i/home/hugin/Documents/haskell/stack-lh-min/.stack-work/dist/x86_64-linux-tinfo6/Cabal-3.2.1.0/build/stack-lh-min-exe","-i/home/hugin/Documents/haskell/stack-lh-min/app","-i/home/hugin/Documents/haskell/stack-lh-min/.stack-work/dist/x86_64-linux-tinfo6/Cabal-3.2.1.0/build/stack-lh-min-exe/autogen","-i/home/hugin/Documents/haskell/stack-lh-min/.stack-work/dist/x86_64-linux-tinfo6/Cabal-3.2.1.0/build/global-autogen","-i/home/hugin/Documents/haskell/stack-lh-min/.stack-work/dist/x86_64-linux-tinfo6/Cabal-3.2.1.0/build/stack-lh-min-exe/stack-lh-min-exe-tmp","-stubdir=/home/hugin/Documents/haskell/stack-lh-min/.stack-work/dist/x86_64-linux-tinfo6/Cabal-3.2.1.0/build","-package-id=base-4.14.3.0","-Wall","-Wcompat","-Widentities","-Wincomplete-record-updates","-Wincomplete-uni-patterns","-Wmissing-export-lists","-Wmissing-home-modules","-Wpartial-fields","-Wredundant-constraints","-rtsopts","-with-rtsopts=-N","-optP-include","-optP/home/hugin/Documents/haskell/stack-lh-min/.stack-work/ghci/65f48719/cabal_macros.h","-ghci-script=/tmp/haskell-stack-ghci/902b433f/ghci-script","-package-db","/home/hugin/Documents/haskell/stack-lh-min/.stack-work/install/x86_64-linux-tinfo6/7c444beeea79842f1b013590dc23a79fb9a40e04eb4aa3bf7bdcf007de05a149/8.10.7/pkgdb","-package-db","/home/hugin/.stack/snapshots/x86_64-linux-tinfo6/7c444beeea79842f1b013590dc23a79fb9a40e04eb4aa3bf7bdcf007de05a149/8.10.7/pkgdb","-package-db","/home/hugin/.stack/programs/x86_64-linux/ghc-tinfo6-8.10.7/lib/ghc-8.10.7/package.conf.d"], componentRoot = "/home/hugin/Documents/haskell/stack-lh-min", componentDependencies = ["stack-lh-min.cabal","package.yaml","stack.yaml"]},"/home/hugin/.stack/programs/x86_64-linux/ghc-tinfo6-8.10.7/lib/ghc-8.10.7")
2022-09-05T14:32:48.291672Z | Info | Interface files cache directory: /home/hugin/.cache/ghcide/main-ff5c5a8fe83d845319b93f906dddc7472085f147
2022-09-05T14:32:48.291852Z | Info | Making new HscEnv. In-place unit ids: [main]
2022-09-05T14:32:48.297784Z | Debug | New component cache HscEnvEq: (([],Just HscEnvEq 5),fromList [("package.yaml",Just 2022-09-05 14:29:52.273424872 UTC),("stack-lh-min.cabal",Just 2022-09-05 14:29:52.273424872 UTC),("stack.yaml",Just 2022-09-05 14:29:52.273424872 UTC)])
2022-09-05T14:32:48.299448Z | Debug | Known files updated:
  fromList [(TargetFile NormalizedFilePath "/home/hugin/Documents/haskell/stack-lh-min/app/Main.hs",fromList ["/home/hugin/Documents/haskell/stack-lh-min/app/Main.hs"]),(TargetModule Paths_stack_lh_min,fromList ["/home/hugin/Documents/haskell/stack-lh-min/.stack-work/dist/x86_64-linux-tinfo6/Cabal-3.2.1.0/build/stack-lh-min-exe/autogen/Paths_stack_lh_min.hs"])]
2022-09-05T14:32:48.299635Z | Debug | Finished build session
AsyncCancelled
2022-09-05T14:32:48.299683Z | Debug | Restarting build session due to new component
Action Queue: [User TypeCheck]
Keys: [GhcSessionIO; , GetKnownTargets; ]
Aborting previous build session took 0.00s 
2022-09-05T14:32:48.299933Z | Debug | Getting hlint ideas for  NormalizedFilePath "/home/hugin/Documents/haskell/stack-lh-min/app/Main.hs"
2022-09-05T14:32:48.456384Z | Debug | Finished: User TypeCheck Took: 0.16s
2022-09-05T14:32:48.457724Z | Debug | Finished: GetHie Took: 0.00s
2022-09-05T14:32:48.457926Z | Debug | Finished: GenerateCore Took: 0.00s

Completed (1 file worked, 0 files failed)
```

**Comment by ashkann** [comment]
After reading some more on how `stack.yaml` and `extra-deps` work I'm even more confused :
```yaml
extra-deps:
 - acme-missiles-0.3
``` 
> what you're saying to Stack is: "if at any point you find that you need to build the acme-missiles package, please use version 0.3". You are not saying "please build acme-missiles now." You are also not saying "my package depends on acme-missiles." You are simply making it available should the need arise.

So basically listing LH packages in `stack.yaml` under `extra-deps` should have exactly zero effects *unless* you are also listing them in `pckage.yaml` or maybe transiently referring to them somehow(?) which is not the case here as in seen in the example repo since it only depends on `base` (not `liquid-base`) 🤔

**Comment by ashkann** [comment]
Switched to cabal and it works. Maybe this is not a HLS issue ?

**Comment by fendor** [comment]
Might still be an HLS issue, keep open until we know what's up.

**Comment by ashkann** [comment]
There is the fact though that just listing the LH packages in `extra-deps` breaks the setup. Asked stack people and they also agree this shouldn't be. So wether or not this is part an HLS issue as well, at any rate stack shouldn't break things when a dependency is listed but never used (which is the case here).

**Comment by fendor** [comment]
Resolved as blocked on stack.

**Comment by 89oinotna** [comment]
I have the same problem, is there any hope to get this working with stack without having to use cabal?

**Comment by fendor** [comment]
To fix this, you should probably raise an issue with stack and see what the maintainers think about it.

**Comment by igormoreno** [comment]
I'm getting this exact same problem now using [this](https://github.com/ucsd-progsys/lh-plugin-demo/blob/main/stack/stack-9.0.2.yaml) stack file from https://github.com/ucsd-progsys/lh-plugin-demo/
